### ISSUE TITLE
Data Leakage Issue: Scaler Fitted on Entire Dataset Before Train/Test Split

### ISSUE THREAD

**Issue by Dys-347** [daxin007/ARMD#5]
I found a critical data leakage issue in the CustomDataset class that could significantly impact model evaluation results. The normalization scaler is fitted on the entire raw dataset before splitting into train and test sets, which means the test set statistics leak into the training process.
Location
File: CustomDataset.__init__() and CustomDataset.read_data()
Lines:

```
self.rawdata, self.scaler = self.read_data(data_root, self.name)  # Line ~24
# ...
self.data = self.__normalize(self.rawdata)  # Line ~34
train, inference = self.__getsamples(self.data, proportion, seed)  # Line ~35

```
Problem Details
Current Implementation Flow:

read_data() loads the entire dataset
Scaler (StandardScaler or MinMaxScaler) is fitted on ALL data
Data is normalized using this scaler
Then data is split into train/test using __getsamples() → divide()

**Comment by Kareneveve** [comment]
I also think this issue will seriously impact  the evaluation results. Have you already fixed and tested this problem? Could you please share me of the revised experimental results?

i have tried only use training data to fitted Scaler, but i got a bad result : (

**Comment by albertcity** [comment]
<img width="1204" height="540" alt="Image" src="https://github.com/user-attachments/assets/364fb57a-8c6d-45f1-8d41-5a05f4ad400b" />
Correct me if I am wrong. I found that there are much more data leakage according to the official config.yaml, in which the args `proportion` of train_dataset and test_dataset are different. 
In the above case, train_dataset's proportion=0.8, it will contains the samples  of  range 0-0.8; For test_datasets's proportion=0.2, it will contains range 0.2-1.0. Therefore, There exists a data leakage for samples in range 0.2-0.8.
